problem to use this extension in oommf 2.1a

Hi,

I have tried to use your package, but I obtain an error: " Error thrown from inside "Oxs_Run" --- Oxs_Ext error inside Run() function Oxs_TimeDruiver --- Oxs_ext ERROR in object Anv_SpintEvolve_3d:: Programming error: Implementation of Oxs_TimeEvolver::Step(const Oxs_timeDrive*, Oxs_ConstKey<Oxs_SimState>, Oxs_DriverStepInfo,Oxs_Key<Oxs_SimState>&) not provided.

I am using oommf 2.1a. I have considered the following implementation:

Specify Anv_SpinTEvolve_3d {
  do_precess 1
  gamma_LL 2.21e5
  method rk4
  alpha 0.01
  beta 0.04

  u {Oxs_ScriptVectorField {
               atlas :disk
               script Current_u
               script_args rawpt
	       norm  1
              
            }}
	
}


Specify Oxs_TimeDriver [subst {
 basename J=0.85e12
 evolver Anv_SpinTEvolve_3d
 stopping_dm_dt 0.001
 mesh :mesh
 m0 :file1
 Ms { Oxs_VecMagScalarField {
   field :file1
}}
}]


Destination savefile mmArchive

Schedule Oxs_TimeDriver::Magnetization savefileStep 100
Schedule DataTable arx savefile 100

What could this error be caused by?

thanks

Hi, I have tried to compile with oommf 2.0b0. But I get the following error:

tclsh oommf.tcl pimake

/home/Mike/oommf/app/oxs/local/spintevolve_3d.cc: In member function ‘virtual OC_BOOL Anv_SpinTEvolve_3d::Step(const Oxs_TimeDriver*, Oxs_ConstKey<Oxs_SimState>, const Oxs_DriverStepInfo&, Oxs_Key<Oxs_SimState>&)’:
/home/Mike/oommf/app/oxs/local/spintevolve_3d.cc:1842:20: error: ‘const struct Oxs_DriverStepInfo’ has no member named ‘GetCurrentAttemptCount’; did you mean ‘current_attempt_count’?
 1842 |       && step_info.GetCurrentAttemptCount()==0)) {
      |                    ^~~~~~~~~~~~~~~~~~~~~~
      |                    current_attempt_count
<3859> pimake 2.0b0  panic:
child process exited abnormally
1 Error(s) detected

--- a/README.md
+++ b/README.md
@@ -15,6 +15,8 @@ version of the extension can be seen in
 it is distributed as part of OOMMF and taken from the OOMMF source distribution (further details are availble under https://math.nist.gov/oommf/contrib/oxsext/).
 The modified version supports arbitrary current directions (`3d`).
 
+The current `main` is compatible with OOMMF 2.1a0, for OOMMF 2.0b use the [release 1.0](https://github.com/oommf-extensions/anv-spintevolve-3d/releases/tag/v1.0)
+
 Support in [Ubermag](https://ubermag.github.io) will be provided in the next release.
 
 ## Installation options

--- a/src/spintevolve_3d.cc
+++ b/src/spintevolve_3d.cc
@@ -804,7 +804,7 @@ void Anv_SpinTEvolve_3d::NegotiateTimeStep
   if(stepsize<timestep_lower_bound) stepsize = timestep_lower_bound;
 
   // Negotiate with driver over size of next step
-  driver->FillState(cstate,nstate);
+  driver->FillStateMemberData(cstate,nstate);
   UpdateTimeFields(cstate,nstate,stepsize);
 
   // Update iteration count
@@ -1823,7 +1823,7 @@ void Anv_SpinTEvolve_3d::AdjustStepHeadroom(OC_INT4m step_reject)
 OC_BOOL
 Anv_SpinTEvolve_3d::Step(const Oxs_TimeDriver* driver,
                       Oxs_ConstKey<Oxs_SimState> current_state_key,
-                      const Oxs_DriverStepInfo& step_info,
+                      Oxs_DriverStepInfo& step_info,
                       Oxs_Key<Oxs_SimState>& next_state_key)
 {
   const OC_REAL8m bad_energy_cut_ratio = 0.75;

--- a/src/spintevolve_3d.h
+++ b/src/spintevolve_3d.h
@@ -276,7 +276,7 @@ public:
   virtual  OC_BOOL
   Step(const Oxs_TimeDriver* driver,
        Oxs_ConstKey<Oxs_SimState> current_state,
-       const Oxs_DriverStepInfo& step_info,
+       Oxs_DriverStepInfo& step_info,
        Oxs_Key<Oxs_SimState>& next_state);
   // Returns true if step was successful, false if
   // unable to step as requested.

